### DISCUSSION TITLE
Use GitHub tarballs rather than zipballs

### DISCUSSION THREAD

**Discussion by abitrolly** [Homebrew/discussions#3665] [Tap maintenance and Homebrew development] [answered]
`brew doctor` says to use tarballs. But why? `.zip` allows random access, so one don't need to unpack the whole file to inspect the contents.
```
  * 4: col 3: Use GitHub tarballs rather than zipballs (url is https://github.com/sorbet/sorbet/archive/refs/tags/0.5.10439.20220916051958-58983475a.zip).
```

**Comment by carlocab** [comment, marked as answer]
Tarballs can be about 40% smaller than zipballs, and we pretty much never need random access when building formulae. Compare, for example, [llvmorg-15.0.0.zip](https://github.com/llvm/llvm-project/archive/refs/tags/llvmorg-15.0.0.zip) and [llvmorg-15.0.0.tar.gz](https://github.com/llvm/llvm-project/archive/refs/tags/llvmorg-15.0.0.tar.gz).

**Reply by abitrolly** [reply]
If download size if critical, does `brew` promote .7z format when it is available? Or can it actually compare which is smaller?

**Reply by SMillerDev** [reply]
because macOS does not come with the tools to extract that, so that would mean extra dependencies are needed to use it. Tarballs are the smallest natively supported format.

**Reply by abitrolly** [reply]
Should not be a problem if some 7z unpacking lib comes with `brew` installation.

**Reply by SMillerDev** [reply]
But it won't since hardly anyone ever uploads `.7z` files containing sources.

**Reply by abitrolly** [reply]
I guess that is chicken and egg problem. While nobody wants better compressed sources, nobody produces them.

**Reply by SMillerDev** [reply]
Or `7z` is not a standard file format that anyone uses because it requires additional software beyond what operating systems normally ship. File size is the deciding factor in tar vs zip, not in all matters.
